inst/include/mode/solver.hpp

```diff
@@ -99,13 +99,38 @@ class solver {
     }
   }
 
-  void reset(double t, std::vector<double> y) {
-    stepper_.reset(t, y);
+  void reset(double t,
+             std::vector<double> y,
+             bool reset_step_size) {
+    set_time(t);
+    stepper_.set_state(t, y);
+    if (reset_step_size) {
```

Comment:
It feels like some of this logic is duplicated in mode.hpp; perhaps if we exposed the primative reset methods here (reset step size, reset statistics, set state, set time) and not support any compound reset here that may tidy the logic up?